symbolic-debuginfo/src/dwarf.rs

```diff
@@ -1342,6 +1344,7 @@ impl<'s> Iterator for DwarfFileIterator<'s> {
 pub struct DwarfFunctionIterator<'s> {
     units: DwarfUnitIterator<'s>,
     functions: std::vec::IntoIter<Function<'s>>,
+    seen_ranges: BTreeSet<(u64, u64)>,
```

Comment:
On second thought, MSVC may output similar duplicate entries. Can you check whether this is the case and we need to port this over, too? 

Also, I'm curious on the runtime impact on this solution. The function iterator is already quite slow due to its sorting, but we should be careful adding another _n log n_ contributor.